viewing_party/party.py

```diff
@@ -1,23 +1,190 @@
+    for movies in user_data["friends"]:
+        total_friends_watched += movies["watched"]
+    
+    #loops through movies in the total_friends_watched list to compare (in the if statement) if the movie is not in the user's watched list and hasn't appeared in the unique_movies_list (checks for duplicates) to then append the unique movie
+    for unique_movies in total_friends_watched:
+        if (unique_movies not in user_data["watched"]) and (not unique_movies in unique_movies_list):
+            unique_movies_list.append(unique_movies)
+    
+    return unique_movies_list
 
-        
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    # list titles of movies the user has already watched 
+    user_watched_list = []
+    for movie in user_data["watched"]:
+        user_watched_list.append(movie["title"])
+    
+    # list that has rec movies
+    recommendations = []
+
+    # loop through each friend
+    for friend in user_data["friends"]:
+        # loop through their moviees
+        for friend_movie in friend["watched"]:
+            # check if user hasn't watched the movie and if host (streaming services) is also under user
+            if (friend_movie["title"] not in user_watched_list and friend_movie["host"] in user_data["subscriptions"] and friend_movie not in recommendations):
+                recommendations.append(friend_movie)
+
+    return recommendations
+
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    #get user's most frequently watched genre
+    genre_count = {}
+    for movie in user_data["watched"]:
+        genre = movie["genre"]
+        if genre in genre_count:
+            genre_count[genre] += 1
+        else:
+            genre_count[genre] = 1
+
+# Find the most frequent genre
+    most_frequent_genre = None
+    max_count = 0
+
+    for genre, count in genre_count.items():
+        if count > max_count:
+            most_frequent_genre = genre
+            max_count = count
+    if most_frequent_genre is None:
+        return []
+    
+    # list of movies user has already watched
+    user_watch = []
+    for movie in user_data["watched"]:
+        user_watch.append(movie["title"])
+    
+    #movie needs to be added to recommeneded list if they haven't watched it but one of their friends watched it
+    recommended_list = []
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            if friend_movie["genre"] == most_frequent_genre and friend_movie["title"] not in user_watch and friend_movie not in recommended_list:
+                recommended_list.append(friend_movie)
+    #returns list of recommended movies
+    return recommended_list
+
+def get_rec_from_favorites(user_data):

```

Comment:
How could we reuse our prior function `get_unique_watched` to simplify the code here?